lib/resque/scheduler.rb

```diff
@@ -208,16 +210,24 @@ def enqueue_delayed_items_for_timestamp(timestamp)
         end
       end
 
+      def enqueue(config)
+        fallback = lambda do |e|
+          Resque::Scheduler::Plugin.run_failure_hooks(failure_handler, config, e)
+        end
+
+        handle_errors(fallback) { enqueue_from_config(config) }
+      end
+
       def handle_shutdown
         exit if @shutdown
         yield
         exit if @shutdown
       end
 
-      def handle_errors
+      def handle_errors(fallback)
```

Comment:
do we really need this method? seems unnecessary now. I would just inline this code in the `enqueue` method.